### DISCUSSION TITLE
Detect login attemps

### DISCUSSION THREAD

**Discussion by ddellarocca** [emqx/emqx#12738] [Q&A]
Hi,
Is there a way to detect failed login attempts? This could be useful for detecting brute force logins from external actors.
I've tried to log in with a user that wasn't present in the built-in database authenticator with EMQX 5.5.1 but that action wasn't recorded in logs by EMQX.

**Comment by zmstone** [comment]
Hi @ddellarocca 
there is `$events/client_check_authz_complete` event which can be intercepted by rule-engine and send such events to a database for auditing purposes.
i believe we should do the same for `authn`, feel free to create a feature request.

**Reply by ddellarocca** [reply]
Done, thanks!
